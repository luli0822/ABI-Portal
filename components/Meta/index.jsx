import React from 'react';
import PropTypes from 'prop-types';
import ContentfulMeta from '../../contentful/ContentfulMeta';

function Meta({ pageId, siteDomain }) {
	return (
		<>
			{/* Need to pass metaTitle  prop to ContentfulMeta  to override contentful the default Title */}
			<ContentfulMeta siteDomain={siteDomain} pageId={pageId} />
		</>
	);
}

Meta.propTypes = {
	pageId: PropTypes.string.isRequired,
	siteDomain: PropTypes.string.isRequired,
};

export default Meta;
