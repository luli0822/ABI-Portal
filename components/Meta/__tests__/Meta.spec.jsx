import React from 'react';
import { shallow } from 'enzyme';
import Meta from '../index';
import {
	mockLang,
	mockPathMap,
	mockLinkedEntries,
	mockLinkedAssets,
	mockSiteEntry,
	mockFieldOrFallback,
} from '../../../contentful/__mocks__/SiteDataContext';

jest.mock('react', () => {
	const ActualReact = require.requireActual('react');
	return {
		...ActualReact,
		useContext: () => ({
			pathMap: mockPathMap,
			fieldOrFallback: mockFieldOrFallback,
			linkedEntries: mockLinkedEntries,
			linkedAssets: mockLinkedAssets,
			siteEntry: mockSiteEntry,
			lang: mockLang,
		}),
	};
});

const props = {
	pageId: 'site-home-page',
	siteDomain: '//www.domain.ca',
};

describe('<Meta />', () => {
	it('renders correctly', () => {
		// eslint-disable-next-line react/jsx-props-no-spreading
		expect(shallow(<Meta {...props} />).debug()).toMatchSnapshot();
	});
});
