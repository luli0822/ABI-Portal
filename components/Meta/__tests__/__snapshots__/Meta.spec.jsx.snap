// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Meta /> renders correctly 1`] = `
"<Fragment>
  <ContentfulMeta siteDomain=\\"//www.domain.ca\\" pageId=\\"site-home-page\\" metaTitle={[undefined]} />
</Fragment>"
`;
